tests/semantic_analyser.cpp

```diff
@@ -528,12 +528,41 @@ TEST(semantic_analyser, call_kaddr)
 
 TEST(semantic_analyser, call_uaddr)
 {
-  test("kprobe:f { uaddr(\"counter\"); }", 0);
-  test("kprobe:f { uaddr(\"github.com/golang/glog.severityName\"); }", 0);
```

Comment:
This test was to make sure that Go symbols with non-alpha-numeric characters are allowed by `check_symbol()`. Now that we're actually resolving symbols in the semantic analyser can `check_symbol` just be removed entirely?

Reply:
Actually maybe we should keep this as a regression test to make sure no one breaks Go symbols in the future, even if we remove `check_symbol`. What do you think?

Reply:
I'll add them back in.